File: cpp/AnonymousClass.cpp
#include <iostream>

class Dog {
    private: 
        int age = 10;
    public: 
        void move() {
            std::cout << "The dog walks! \n";
        }
};

class : private Dog {
    public:
        void move() {
            std::cout << "The dog flies!\n";
        }
} superDog; // Variabel class ini hanya bisa dibuat disini


class {
    public: 
        void eat() {
            std::cout << "The cat eats!\n";
        }
} cat; // Variabel class ini hanya bisa dibuat disini

int main() {
    Dog regulerDog;
    regulerDog.move(); // The dog walks!
    std::cout << typeid(regulerDog).name() << "\n"; // 3Dog ==> class Dog

    superDog.move(); // The dog flies!
    std::cout << typeid(superDog).name() << "\n"; // 9._anon_62 ==> Anonymous class

    cat.eat(); // The cat eats!
    std::cout << typeid(cat).name() << "\n"; // 9._anon_63 ==> Anonymous class
}

File: python/AnonymousClass.py
from traceback import print_tb


class Dog:
    x_value = 10
    def move(self):
        print("The dog walks!")

obj1 = Dog()
obj1.move()
print(type(obj1))
print(obj1.x_value)

obj2 = type('Anonymous Class', (Dog, object), {'move': lambda self: print("The dog flies!")})
obj2.move(obj2)
print(type(obj2))
print(obj2.x_value)